src/lib.rs

```diff
@@ -294,7 +294,10 @@ pub struct Package {
     pub authors: Vec<String>,
     /// An opaque identifier for a package
     pub id: PackageId,
-    source: Option<String>,
+    /// The source of the package, e.g. 
+    /// "registry+https://github.com/rust-lang/crates.io-index"
+    /// or `None` for path dependencies or workspace members
+    pub source: Option<String>,
```

Comment:
I think the reason it was private is that I was wondering whether we'd want to have a structured way to access this field. Can you create a newtype around `String`, keeping the `String` field private and only offering it via a `Display` implementation?

Reply:
Not a bad idea but I fear slightly inconsistent. E.g. source in Dependency is still an `Option<String>`